net/error.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2019 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package net
+
+type ConnectionError struct {
```

Comment:
Shall we also implement the `error` interface so we can treat it as a golang error? https://go.dev/blog/error-handling-and-go

Reply:
Maybe, but I'd rather leave that discussion separate from this PR.